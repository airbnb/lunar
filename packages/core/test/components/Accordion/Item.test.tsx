import React from 'react';
import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import AccordionItem from '../../../src/components/Accordion/Item';

describe('<AccordionItem />', () => {
  it('renders a button', () => {
    const wrapper = shallowWithStyles(
      <AccordionItem id=".0" index={0} title="Title" onClick={() => {}} />,
    );

    expect(wrapper.find('button')).toHaveLength(1);
  });

  it('renders a section', () => {
    const wrapper = shallowWithStyles(
      <AccordionItem id=".0" index={0} title="Title" onClick={() => {}} />,
    );

    expect(wrapper.find('section')).toHaveLength(1);
  });

  it('renders the title as a string', () => {
    const wrapper = shallowWithStyles(
      <AccordionItem id=".0" index={0} title="unique string" onClick={() => {}} />,
    );

    expect(wrapper.find('button').html()).toMatch('unique string');
  });

  it('renders the title as a component', () => {
    function CustomTitleComponent() {
      return null;
    }

    const wrapper = shallowWithStyles(
      <AccordionItem id=".0" index={0} title={<CustomTitleComponent />} onClick={() => {}} />,
    );

    expect(wrapper.find(CustomTitleComponent)).toHaveLength(1);
  });

  it('renders children', () => {
    const child = <div>Foo</div>;

    const wrapper = shallowWithStyles(
      <AccordionItem id=".0" index={0} title="Title" onClick={() => {}}>
        {child}
      </AccordionItem>,
    );

    expect(wrapper.contains(child)).toBe(true);
  });

  it('renders bordered', () => {
    const wrapper = shallowWithStyles(
      <AccordionItem bordered id=".0" index={0} title="Title" onClick={() => {}} />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders expanded', () => {
    const wrapper = shallowWithStyles(
      <AccordionItem expanded id=".0" index={0} title="Title" onClick={() => {}} />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('triggers `onClick` when clicked', () => {
    const spy = jest.fn();
    const wrapper = shallowWithStyles(
      <AccordionItem id=".0" index={1} title="Title" onClick={spy} />,
    );

    wrapper
      .find('button')
      .at(0)
      .simulate('click');

    expect(spy).toHaveBeenCalledWith(1);
  });
});
