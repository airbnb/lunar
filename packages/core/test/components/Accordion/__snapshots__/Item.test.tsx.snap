// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionItem /> renders bordered 1`] = `
<div
  className="item_14cfb9u-o_O-item_bordered_hg23ez"
>
  <button
    aria-controls="accordion-body-.0"
    aria-selected={false}
    className="title_1sndfdc"
    id="accordion-title-.0"
    onClick={[Function]}
    role="tab"
    tabIndex={0}
    type="button"
  >
    <span
      className="titleText_irsnq7"
    >
      Title
    </span>
    <IconChevronRight
      decorative={true}
      size={24}
    />
  </button>
  <section
    aria-hidden={true}
    aria-labelledby="accordion-title-.0"
    className="body_1votmn2"
    id="accordion-body-.0"
    role="tabpanel"
  />
</div>
`;

exports[`<AccordionItem /> renders expanded 1`] = `
<div
  className="item_14cfb9u"
>
  <button
    aria-controls="accordion-body-.0"
    aria-selected={true}
    className="title_1sndfdc"
    id="accordion-title-.0"
    onClick={[Function]}
    role="tab"
    tabIndex={0}
    type="button"
  >
    <span
      className="titleText_irsnq7"
    >
      Title
    </span>
    <IconChevronDown
      decorative={true}
      size={24}
    />
  </button>
  <section
    aria-hidden={false}
    aria-labelledby="accordion-title-.0"
    className="body_1votmn2-o_O-body_expanded_uhlm2"
    id="accordion-body-.0"
    role="tabpanel"
  />
</div>
`;
